### PR TITLE
Ignore nonexistent system temporary directory when a valid cache directory is provided via options

### LINKED ISSUE #65: Cache type not usable if global sys_tmp_dir not writeable
### BC Break Report

|    Q        |   A
|------------ | ------
| Version     | 2.3.x-dev

#### Summary
When you define an own cache_dir, the system always tries the default sys_get_temp_dir first, even though you are not using it. If anything is wrong with it (like no write access), you cannot create a FileSystem Adapter at all.

#### Previous behavior

The sys_get_temp_dir() was only used in case that no other dir was defined.

#### Current behavior

I've provided the following config for the filesystem:

```
'name'    => 'filesystem',
'adapter' => 'filesystem',
'options' => [
    'cache_dir' => 'data/cache/',
    'ttl'       => '3600'
],
'plugins' => [
    [
        'name'    => 'exception_handler',
        'options' => [
            'throw_exceptions' => true,
        ],
    ],
    [
        'name' => 'Serializer'
    ]
]
```

As you can see, the **cache_dir** is provided in the settings. 
However, since the new version, the FileSystemOptions always calls the setCacheDir with the parameter `null`:
 `$this->setCacheDir(null);` (`FileSystemOptions.php:115`)

This results in a fallback to sys_get_temp_dir(), which will then be checked and - on my test maschine - was not writable. The Adapter throws and cannot be created then, though right after that, it would overwrite the CacheDir with the correct and working path.

#### How to reproduce
Define your `sys_temp_dir` in your `php.ini` to a non-existing or not-writable path. 
Then try to create an adapter with a valid `cache_dir` configuration. 
It will not work and fails with one exception of the method `normalizeCacheDirectory`

### PR DESCRIPTION
**Pull request by bcremer**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | yes
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

Fix for https://github.com/laminas/laminas-cache-storage-adapter-filesystem/issues/65

### CODE REVIEW

File: .laminas-ci.json [hunk 1]
```diff
@@ -1,2 +1,12 @@
 {
+  "additional_checks": [
```
Comment (Ocramius) [inline .laminas-ci.json]: Instead of adding a whole CI run, can we have a single test that uses `exec` on a dummy script, perhaps? 🤔

**Review by Ocramius — APPROVED** [review]
LGTM meanwhile 👍

File: .laminas-ci.json (hunk 1 above)
Comment (boesing) [reply .laminas-ci.json]: Do you expect that test to check the exit code or how do we ensure that exactly the non-writable directory leads to the issue and not something else?

Comment (boesing) [reply .laminas-ci.json]: I moved the check into a dedicated script which is now executed from a test method.

**Comment by boesing** [comment]
@bcremer FYI, it seems your GPG key is not stored on github and thus your commit is marked "unverified"

**Pull request merged**